src/main/app/src/components/Purchasing.js

```diff
@@ -41,25 +54,163 @@ const orderRows = [
   { id: 4, vendorName: 'Vendor1', items: '????', timestamp: "01/31/2021", cost: "$100" }
 ];
 
-const LoadedView = () => {
+function TextMaskCustom(props) {
+  const { inputRef, ...other } = props;
+
+  return (
+    <MaskedInput
+      {...other}
+      ref={(ref) => {
+        inputRef(ref ? ref.inputElement : null);
+      }}
+      mask={['(', /[1-9]/, /\d/, /\d/, ')', ' ', /\d/, /\d/, /\d/, '-', /\d/, /\d/, /\d/, /\d/]}
+      placeholderChar={'\u2000'}
+      showMask
+    />
+  );
+}
+
+TextMaskCustom.propTypes = {
+  inputRef: PropTypes.func.isRequired,
+};
+
+const PhoneNumberField = () => {
+  const [values, setValues] = React.useState({
+    textmask: '(  )    -    ',
+    numberformat: '1320',
+  });
+
+  const handleChange = (event) => {
+    setValues({
+      ...values,
+      [event.target.name]: event.target.value,
+    });
+  };
+  return (
+    <FormControl>
+        <InputLabel>Phone Number</InputLabel>
+        <Input
+          value={values.textmask}
+          onChange={handleChange}
+          name="textmask"
+          inputComponent={TextMaskCustom}
+        />
+      </FormControl>
+  );
+};
+
+const ProvinceDrawer = () =>{
+  const [province, setProvince] = React.useState('');
+
+  const handleChange = (event) => {
+    setProvince(event.target.value);
+  };
+  return(
+    <div  style={{ width: "250px" }}>
+      <FormControl>
+        <Select
+          value={province}
+          onChange={handleChange}
+        >
+          <MenuItem value={"QC"}>QC</MenuItem>
+          <MenuItem value={"ON"}>ON</MenuItem>SS
+        </Select>
+      </FormControl>
+    </div>
+  );
+};
+
+function LoadedView() {
+  const [open, setOpen] = React.useState(false);
+
+  const handleClickOpen = () => {
+    setOpen(true);
+  };
+
+  const handleClose = () => {
+    setOpen(false);
+  };
+
+  const AddCustomerDialogBox = () => {
+    return (
+      <Dialog open={open} onClose={handleClose} fullWidth maxWidth='sm'>
+        <DialogTitle>Add New Vendor</DialogTitle>
+        <DialogContent>
+          <DialogContentText>
+            Please Do Not Leave Any Fields Empty!
+          </DialogContentText>
+          <InputLabel>Vendor Name</InputLabel>
+          <TextField
+            autoFocus
+            margin="dense"
+            id="name"
+            type="text"
+            />
+          <InputLabel>Address</InputLabel>
+          <TextField
+            autoFocus
+            margin="dense"
+            type="text"
+            multiline
+            rows={2}
```

Comment:
The text in the "Address" field is too high up. Can you place it right against the line? I think this is because of line 155 you put field as 2 rows (should be 1 row)

Reply:
Yeah I was experimenting with Multi Row text fields in the case of long addresses. Will switch to 1 row